src/Build/Collections/RetrievableEntryHashSet/RetrievableValuedEntryHashSet.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+#nullable disable
+
+namespace Microsoft.Build.Collections
+{
+    /// <inheritdoc />
+    [DebuggerTypeProxy(typeof(HashSetDebugView<>))]
+    [DebuggerDisplay("Count = {Count}")]
+    [Serializable()]
```

Comment:
Can we remove this?

Is this class going throgh any serialization? If yes - wehre?

Reply:
Fixed.